batch/batch/worker/worker.py

```diff
@@ -1387,6 +1441,19 @@ async def setup_io(self):
         assert self.disk is None, self.disk
         os.makedirs(self.io_host_path())
 
+    async def run_container(self, container: Container, task_name: str):
+        async with container.run():
+            with container._step('uploading_log'):
+                assert self.worker.file_store
+                await self.worker.file_store.write_log_file(
+                    self.format_version,
+                    self.batch_id,
+                    self.job_id,
+                    self.attempt_id,
+                    task_name,
+                    await container.get_log(),
+                )
```

Comment:
I think this might be a bit *too* clever a use of a context manager. What do you think of just adding a parameter to the Container constructor which is an async function `upload_log`?